packages/camera/camera_avfoundation/ios/camera_avfoundation/Sources/camera_avfoundation/FLTCam.m

```diff
@@ -278,9 +275,33 @@ - (instancetype)initWithConfiguration:(nonnull FLTCamConfiguration *)configurati
 
   [self updateOrientation];
 
+  // Handle video and audio interruptions and errors.
+  // https://github.com/flutter/flutter/issues/151253
+  for (NSObject<FLTCaptureSession> *session in @[ _videoCaptureSession, _audioCaptureSession ]) {
+    [NSNotificationCenter.defaultCenter addObserver:self
+                                           selector:@selector(captureSessionWasInterrupted:)
+                                               name:AVCaptureSessionWasInterruptedNotification
+                                             object:session.captureSession];
+
+    [NSNotificationCenter.defaultCenter addObserver:self
+                                           selector:@selector(captureSessionRuntimeError:)
+                                               name:AVCaptureSessionRuntimeErrorNotification
```

Comment:
when will run time error happen?

Reply:
As is in the description "for example when there is an incoming phone call and the camera starts recording with audio". When `startVideoRecording` is called with enabled audio while there is an incoming call (does not need to be answered in my case, ringing is enough).

Reply:
can you put some comments there?